api/proto/config/storage.proto

```diff
@@ -41,6 +42,8 @@ message DatastoreConfig {
 
 	// Applies for some datastore types
 	string region = 6;
+        bytes cipherTextPassword = 7;
```

Comment:
Don't we need to encrupt the apiKey and password?

As above add comments indicating the relationship between the cleartext and encrypted fields and how they are used/deprecated.